test/DotNetty.Codecs.Tests/Base64Test.cs

```diff
@@ -0,0 +1,163 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace DotNetty.Codecs.Tests
+{
+    using System;
+    using System.Security.Cryptography.X509Certificates;
+    using System.Text;
+    using DotNetty.Buffers;
+    using DotNetty.Codecs.Base64;
+    using Xunit;
+
+    public class Base64Test
+    {
+        [Fact]
+        public void TestNotAddNewLineWhenEndOnLimit()
+        {
+            var src = Unpooled.CopiedBuffer(Encoding.ASCII.GetBytes("____abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcde"));
+            var expectedEncoded = Unpooled.CopiedBuffer(Encoding.ASCII.GetBytes("YWJjZGVmZ2hpamtsbW5vcHFyc3R1dnd4eXphYmNkZWZnaGlqa2xtbm9wcXJzdHV2d3h5emFiY2Rl"));
+            var buff = src.Slice(3, src.ReadableBytes - 3);
+            buff.ReadByte();
+            TestEncode(buff, expectedEncoded);
+        }
+
+        [Fact]
+        public void TestSimpleDecode()
```

Comment:
I was talking specifically about length of the input and its content. There might be some edge cases there regarding output length calculation, character mapping, etc. Even doing length from 1 to 10 would already cover a lot of grounds in that sense. The sample I referenced does Cartesian on all the input dimensions. Here it might be just one dimension for length or two if input content is to be varied. Considering you're using "fixed" it won't be unreasonable to test it thoroughly

---
In reply to: [76195354](https://github.com/Azure/DotNetty/pull/150#discussion_r76195354) [](ancestors = 76195354)